Arm/Insts/DPI/Add_sub_imm.lean

```diff
@@ -24,19 +24,21 @@ def exec_add_sub_imm (inst : Add_sub_imm_cls) (s : ArmState) : ArmState :=
                         else
                           imm <<< 12
     let operand1      := read_gpr datasize inst.Rn s
-    let (carry_in, operand2)
+    let carryInAndOperand2
                       := if sub_op then
                           (1#1, ~~~imm)
                         else
                           (0#1, imm)
+    let carry := carryInAndOperand2.fst
+    let operand2 := carryInAndOperand2.snd
     let operand2         := BitVec.zeroExtend datasize operand2
-    let (result, pstate) := AddWithCarry operand1 operand2 carry_in
+    let resultAndPState := AddWithCarry operand1 operand2 carry
```

Comment:
This pattern-match should be fine, right? IIUC, it was only matching on an if that caused problems

Reply:
Fixed, undone.